includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -1036,10 +1037,12 @@ public function display_payment_method_selected( $checkout_session = null ) {
 	 * session object.
 	 *
 	 * @param  object $checkout_session The active checkout session.
+	 * @param  bool   $force            Whether to force the refresh of the checkout session.
+	 *
 	 * @return boolean
 	 */
-	public function has_payment_preferences( $checkout_session = null ) {
-		if ( null === $checkout_session ) {
+	public function has_payment_preferences( $checkout_session = null, $force = false ) {
+		if ( ! $force && null === $checkout_session ) {
```

Comment:
if the $checkout_session is null shouldn't be refreshed? I mean the conditional should be an OR

Reply:
I was getting a kind of infinite loop in my local when a non-logged user opened the checkout page (Checkout Blocks).
I adjusted the parameter name, the idea is to have bypass creating a new session if the existing session is already null, in checkout blocks this was running over and over causing RAM usage issues.

Reply:
I was missundertanding the idea.

Reply:
The description was far from clear 😞